src/plugins/auto/src/auto_schedule.cpp

```diff
@@ -456,32 +456,32 @@ bool AutoSchedule::schedule_to_worker_infer_request(ov::threading::Task pipeline
     std::vector<DeviceInformation> devices;
     // AUTO work mode
     // Devices that fail infer will be removed from the priority list in the callback, need lock here
-    std::unique_lock<std::mutex> lock(m_context->m_fallback_mutex);
-    if (!preferred_device.empty()) {
-        // if the device needed by customer is not ready, need to wait for it
-        wait_actual_compiled_model_ready();
-        devices.push_back(m_compile_context[ACTUALDEVICE].m_device_info);
-        if (!deviceChecker().check_if_device_in_list<DeviceInformation>(preferred_device, devices)) {
-            lock.unlock();
-            OPENVINO_THROW("The preferred device should be the selected device");
-        }
-    } else {
-        // _acceleratorDevice could be the same as _cpuDevice, such as AUTO:CPU
-        if (m_compile_context[FALLBACKDEVICE].m_is_already) {
-            devices.push_back(m_compile_context[FALLBACKDEVICE].m_device_info);
+    {
+        std::lock_guard<std::mutex> lock(m_context->m_fallback_mutex);
+        if (!preferred_device.empty()) {
+            // if the device needed by customer is not ready, need to wait for it
+            wait_actual_compiled_model_ready();
+            devices.push_back(m_compile_context[ACTUALDEVICE].m_device_info);
+            if (!deviceChecker().check_if_device_in_list<DeviceInformation>(preferred_device, devices)) {
+                OPENVINO_THROW("The preferred device should be the selected device");
+            }
         } else {
-            if (m_compile_context[ACTUALDEVICE].m_is_already) {
-                devices.push_back(m_compile_context[ACTUALDEVICE].m_device_info);
+            // _acceleratorDevice could be the same as _cpuDevice, such as AUTO:CPU
```

Comment:
Does this block not need lock protection?

Reply:
this block has been in the scope of the lock protection here.